python/ray/external_storage.py

```diff
@@ -434,6 +436,21 @@ def destroy_external_storage(self):
 _external_storage = NullStorage()
 
 
+class UnstableFileStorage(FileSystemStorage):
+    """This class is for testing with writing failure."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._failure_rate = 0.1
+
+    def spill_objects(self, object_refs, owner_addresses) -> List[str]:
```

Comment:
It doesn't test the partial failure cases right? (e.g., when 5 keys are given, only 3 of them succeed). 

Why don't we also add failure cases where only the first N entires are returned from the spill_objects method? 

```python3
full_failure_rate = 0.1
partial_failure_ratio  = 0.2

r = random.random()
failed = r < self._failure_rate
partial_failed = r < partial_failure_ratio

if failed:
    raise IOERROR
elif partial_failed:
    i = random.choice(list(len(object_refs)))
    return super().spill_objects(object_refs[:i], owner_address)
        
```

something like this.

Reply:
I feel it actually covered that, but it depends on the randomness actually. I'll update it anyway.